components/feature/privatemode/src/main/java/mozilla/components/feature/privatemode/notification/AbstractPrivateNotificationService.kt

```diff
@@ -68,16 +115,38 @@ abstract class AbstractPrivateNotificationService : Service() {
      */
     @ExperimentalCoroutinesApi
     final override fun onCreate() {
-        val id = SharedIdsHelper.getIdForTag(this, NOTIFICATION_TAG)
-        val channelId = ensureNotificationChannelExists(this, NOTIFICATION_CHANNEL, onSetupChannel = {
-            if (SDK_INT >= Build.VERSION_CODES.O) {
-                enableLights(false)
-                enableVibration(false)
-                setShowBadge(false)
-            }
-        })
+        val notificationId = getNotificationId()
+        val channelId = getChannelId()
+        val notification = createNotification(channelId)
+
+        startForeground(notificationId, notification)
+
+        privateTabsScope = store.flowScoped { flow ->
+            flow.map { state -> state.privateTabs.isEmpty() }
+                .ifChanged()
+                .collect { noPrivateTabs ->
+                    if (noPrivateTabs) stopService()
+                }
+        }
+
+        localeScope = store.flowScoped { flow ->
+            flow.mapNotNull { state -> state.locale }
+                .ifAnyChanged { locale ->
```

Comment:
This shouldn't need `ifAnyChanged`. Since the array has only one element, `ifChanged` should be enough here.